benches/benches/bevy_transform/hierarchy/propagation.rs

```diff
@@ -0,0 +1,200 @@
+}
+
+criterion_group!{
+    name = transform_hierarchy_sizes;
+    config = Criterion::default()
+        .warm_up_time(std::time::Duration::from_millis(300))
+        .measurement_time(std::time::Duration::from_secs(5))
+        .sample_size(50)
+        ;
+
+    targets = transform_propagation_sizes
+}
+
+/// Inner transform propagation benchmark function
+/// This version only measures time spent during PostUpdate, therefore removing
+/// the impact of simulating transform updates which happen during the Update schedule.
+fn update_bench_postupdate_only(b: &mut Bencher<WallTime>, &(cfg, enable_update): &(&Cfg, TransformUpdates)) {
+    let (_result, mut app) = build_app(cfg, enable_update);
+
+    // Run Main schedule once to ensure initial updates are done
+    app.update();
+
+    // We want to benchmark the transform updates in the PostUpdate schedule without
+    // benchmarking the update function which is intended to simulate changes to Transform
+    // in a typical game.
+    // Therefore, we simply remove the PostUpdate and Last schedules here in order to
+    // measure the time spent in PostUpdate itself, without the time spent in the
+    // schedules before PostUpdate (PreUpdate, Update, ...) and the schedules after
+    // PostUpdate (only Last currently).
+    // If the schedules that are part of main change, this logic needs to be changed
+    // accordingly.
+    let mut schedules = app.world.get_resource_mut::<Schedules>().unwrap();
+    let (_, mut postupdate) = schedules.remove_entry(&bevy_app::PostUpdate).unwrap();
+    let (_, mut last) = schedules.remove_entry(&bevy_app::Last).unwrap();
+
+    b.iter_custom(|iters| {
+        let mut total = Duration::ZERO;
+
+        for _i in 0..iters {
+            std::hint::black_box(app.world.run_schedule(bevy_app::Main));
+
+            let start = Instant::now();
+            std::hint::black_box(postupdate.run(&mut app.world));
+            let elapsed = start.elapsed();
+
+            std::hint::black_box({
+                last.run(&mut app.world);
+                app.world.clear_trackers();
+            });
+
+            total += elapsed;
+        }
+
+        total
+    });
+}
+
+/// Inner transform propagation benchmark function
+///
+/// Simpler alternative to update_bench_postupdate_only that is retained here
+/// for future reference. This benchmark includes the time spent simulating
+/// transform updates in the Update schedule which makes the comparison between
+/// noop and transform_updates benchmarks meaningful.
+fn update_bench_reference(b: &mut Bencher<WallTime>, &(cfg, enable_update): &(&Cfg, TransformUpdates)) {
+    let (_result, mut app) = build_app(cfg, enable_update);
+
+    // Run Main schedule once to ensure initial updates are done
+    app.update();
+
+    b.iter(move || { app.update(); });
+
+}
+
+fn inner_update_bench(b: &mut Bencher<WallTime>, bench_cfg: &(&Cfg, TransformUpdates)) {
+    const UPDATE_BENCH_POSTUPDATE_ONLY: bool = false;
```

Comment:
It's a weird way of selecting benchmarks. And inconsistent with how the `transform_init` benchmark works. 

In my opinion "reference" benchmarks should be removed from this PR, since (from my testing) they only add a constant overhead.

But it's OK if they stay in the PR as long as they are consistently declared :P